app/containers/Profile/index.js

```diff
@@ -8,29 +8,262 @@ import React from 'react';
+   */
+
+  uploadImage(event) {
+    try {
+      const file = event.target.files[0];
+      const reader = new window.FileReader();
+
+      reader.onloadend = () => this.props.uploadImageFileDtch(reader.result);
+      return reader.readAsArrayBuffer(file);
+    } catch (err) {
+      return err.message;
+    }
+  }
+
+  /*
+   * @getUserKey: method allows to get unique key from URL
+   * this key gives us opportunity to know info about user
+   * example: /users/user1/ --> got key: user1
+   */
+
+  static getUserKey(str) {
+    let pathname = str.slice(1);
+
+    if (pathname[pathname.length - 1] === '/') {
+      pathname = pathname.slice(0, -1);
+    }
+
+    return pathname.split('/')[1];
+  }
+
   render() {
-    return <div>My profile</div>;
+    const {
+      account,
+      profile,
+      loadingProfile,
+      userKey,
+      cashedProfileImg,
+      editingImgState,
+      editImageDtch,
+      clearImageChangesDtch,
+      chooseLocationDtch,
+      getCitiesListDtch,
+      citiesList,
+      loadingSaveProfile,
+      locale,
+    } = this.props;
+
+    const sendProps = {
+      uploadImage: this.uploadImage,
+      editImage: editImageDtch,
+      clearImageChanges: clearImageChangesDtch,
+      chooseLocation: chooseLocationDtch,
+      getCitiesList: getCitiesListDtch,
+      getCroppedAvatar: this.getCroppedAvatar,
+      cancelChanges: this.componentDidMount,
+      saveProfile: Profile.saveProfile,
+      loadingSaveProfile,
+      citiesList,
+      cashedProfileImg,
+      editingImgState,
+      profile,
+      translations: translationMessages[locale],
+      isOwner: account.eosAccount === userKey,
+    };
+
+    const HelmetTitle = `${(profile.eos && profile.eos.display_name) ||
+      translationMessages[locale][messages.wrongUser.id]} | ${
+      translationMessages[locale][messages.profile.id]
+    }`;
+
+    return (
+      <div className="container">
+        <Helmet>
+          <title>{HelmetTitle}</title>
+          <meta
+            name="description"
+            content={
+              translationMessages[locale][messages.profileDescription.id]
+            }
+          />
+        </Helmet>
+        {!loadingProfile && (
+          <Wrapper>
+            {profile.eos && <ProfileForm sendProps={sendProps} />}
+            {!profile.eos && <NoSuchUser />}
+          </Wrapper>
+        )}
+        {loadingProfile && <Preloader />}
+      </div>
+    );
   }
 }
 
-Profile.propTypes = {};
+Profile.propTypes = {
+  profile: PropTypes.object.isRequired,
+  account: PropTypes.object.isRequired,
+  userKey: PropTypes.string.isRequired,
+  locale: PropTypes.string.isRequired,
+  cashedProfileImg: PropTypes.string.isRequired,
+  blob: PropTypes.string.isRequired,
+  citiesList: PropTypes.array.isRequired,
+  editingImgState: PropTypes.bool.isRequired,
+  loadingSaveProfile: PropTypes.bool.isRequired,
+  loadingProfile: PropTypes.bool.isRequired,
+  getProfileInfoDtch: PropTypes.func.isRequired,
```

Comment:
If we want to use suffix for dispatch functions then let's use full 'Dispatch' or not use suffix at all. Dtch looks confusing to me. It reads as short of 'Detach'.